benches/src/cut_edges.rs

```diff
@@ -201,30 +206,121 @@ fn dispatch_std_threads<T: CoordsFloat>(
     edges: &mut Vec<EdgeIdType>,
     darts: &[DartIdType],
     n_threads: usize,
+    bind_threads: bool,
 ) -> u32 {
     let units: Vec<(u32, [u32; 6])> = edges
         .drain(..)
         .zip(darts.chunks(6))
         .map(|(e, sl)| (e, sl.try_into().unwrap()))
         .collect();
     std::thread::scope(|s| {
-        let mut handles = Vec::new();
-        for wl in units.chunks(1 + units.len() / n_threads) {
-            handles.push(s.spawn(|| {
-                let mut n = 0;
-                wl.iter().for_each(|&(e, new_darts)| {
-                    let mut n_retry = 0;
-                    if map.is_i_free::<2>(e as DartIdType) {
-                        while !process_outer_edge(map, &mut n_retry, e, new_darts).is_validated() {}
-                    } else {
-                        while !process_inner_edge(map, &mut n_retry, e, new_darts).is_validated() {}
+        if bind_threads {
+            #[cfg(not(feature = "thread-binding"))]
+            {
+                unreachable!(); // due to CLI option only being present with the feature
+            }
+            #[cfg(feature = "thread-binding")]
+            {
+                let topology = Topology::new().unwrap();
+                let topology = Arc::new(topology);
+                match get_physical_cores(&topology) {
+                    Ok(mut cores) => {
+                        let n_t = if cores.len() < n_threads {
+                            // don't allow more than one thread per physical core
+                            // this is sane since this branch executes only if we explicitly enable binding
+                            eprintln!(
+                                "W: Less physical cores than logical threads; proceeding with one thread per core ({})",
+                                cores.len()
+                            );
+                            cores.len()
```

Comment:
I do not totally agree with the comment; you should "bind" more than one thread per core. It can be beneficial for exploiting hyper-threading, for example.